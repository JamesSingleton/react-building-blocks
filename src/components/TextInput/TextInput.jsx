import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styled from 'styled-components';

const StyledInput = styled.input`
  min-height: 34px;
  padding: 6px 8px;
  font-size: 1rem;
  line-height: 20px;
  color: #24292e;
  vertical-align: middle;
  background-color: #fff;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d1d5da;
  border-radius: 3px;
  outline: none;
  box-shadow: rgba(27, 31, 35, 0.075) 0px 1px 2px inset;
`;

const TextInput = ({
  size, block, className, ...rest
}) => {
  const classes = classnames(className, 'form-control', {
    'input-block': block,
    'input-sm': size === 'small',
    'input-lg': size === 'large',
  });
  const inputProps = {
    className: classes,
    type: 'text',
    ...rest,
  };
  return <StyledInput {...inputProps} />;
};

TextInput.propTypes = {
  block: PropTypes.bool,
  disabled: PropTypes.bool,
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  name: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'large']),
  value: PropTypes.string.isRequired,
};

TextInput.defaultProps = {
  block: true,
  disabled: false,
  className: '',
  name: '',
  placeholder: 'Placeholder',
  required: false,
  size: 'small',
};

export default TextInput;
