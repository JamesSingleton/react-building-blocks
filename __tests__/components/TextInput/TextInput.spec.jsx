import React from 'react';
import { shallow } from 'enzyme';
import TextInput from '../../../src/components/TextInput';

describe('Text', () => {
  it('renders a small text input component', () => {
    const props = {
      size: 'small',
      id: 'input1',
      onChange: jest.fn(),
      value: 'Input 1',
    };
    const renderedText = shallow(<TextInput {...props} />);
    expect(renderedText).toMatchSnapshot();
  });
  it('renders a large text input component', () => {
    const props = {
      size: 'large',
      id: 'input2',
      onChange: jest.fn(),
      value: 'Input 2',
    };
    const renderedText = shallow(<TextInput {...props} />);
    expect(renderedText).toMatchSnapshot();
  });
});
