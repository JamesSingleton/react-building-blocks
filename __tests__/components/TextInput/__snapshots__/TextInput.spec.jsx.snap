// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders a large text input component 1`] = `
<styled.input
  className="form-control input-block input-lg"
  disabled={false}
  id="input2"
  name=""
  onChange={[MockFunction]}
  placeholder="Placeholder"
  required={false}
  type="text"
  value="Input 2"
/>
`;

exports[`Text renders a small text input component 1`] = `
<styled.input
  className="form-control input-block input-sm"
  disabled={false}
  id="input1"
  name=""
  onChange={[MockFunction]}
  placeholder="Placeholder"
  required={false}
  type="text"
  value="Input 1"
/>
`;
